test/common.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2020, the gRPC project authors. Please see the AUTHORS file
+// for details. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import 'dart:io';
+import 'dart:async';
+import 'package:test/test.dart';
+
+/// Test functionality for both TCP and Unix domain sockets.
+void testTcpAndUds(String name, String host,
```

Comment:
I would make `host` a named parameter with `localhost` default. Then you can omit it in all places (because we mostly pass `localhost`) and still have it flexible. 

`testTcpAndUds(..., {String host = 'localhost'})`